diff --git a/CMakeLists.txt b/CMakeLists.txt
index da7142c9..2c8ebdee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,8 @@ if(POLICY CMP0054)
   cmake_policy(SET CMP0054 NEW)
 endif()
 
+add_definitions(-DGLOG_USE_GLOG_EXPORT)
+
 # ---[ Caffe project
 project(Caffe C CXX)
 
@@ -44,7 +46,7 @@ include(cmake/Dependencies.cmake)
 
 # ---[ Flags
 if(UNIX OR APPLE)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 endif()
 
 caffe_set_caffe_link()
diff --git a/src/caffe/util/io.cpp b/src/caffe/util/io.cpp
index 103427bf..09f32ccd 100644
--- a/src/caffe/util/io.cpp
+++ b/src/caffe/util/io.cpp
@@ -63,7 +63,7 @@ bool ReadProtoFromBinaryFile(const char* filename, Message* proto) {
   CHECK_NE(fd, -1) << "File not found: " << filename;
   ZeroCopyInputStream* raw_input = new FileInputStream(fd);
   CodedInputStream* coded_input = new CodedInputStream(raw_input);
-  coded_input->SetTotalBytesLimit(kProtoReadBytesLimit, 536870912);
+  coded_input->SetTotalBytesLimit(kProtoReadBytesLimit);
 
   bool success = proto->ParseFromCodedStream(coded_input);
 
